crates/trigger/src/cli.rs

```diff
@@ -147,29 +156,44 @@ where
         loader: impl Loader + Send + Sync + 'static,
         locked_url: String,
     ) -> Result<Executor> {
-        let trigger_config =
-            TriggerExecutorBuilderConfig::load_from_file(self.runtime_config_file.clone())?;
+        let runtime_config = self.build_runtime_config()?;
 
         let _sloth_warning = warn_if_wasm_build_slothful();
 
         let mut builder = TriggerExecutorBuilder::new(loader);
         self.update_wasmtime_config(builder.wasmtime_config_mut())?;
 
-        let logging_hooks = StdioLoggingTriggerHooks::new(self.follow_components(), self.log_dir());
+        if runtime_config.state_dir().is_none() {
+            println!("No local state dir; logs and key value data may not be written to disk.");
```

Comment:
It would be good to correlate this with `log_dir`.  Currently if `state_dir` is unset but `log_dir` is set, it seems like Spin will print `Logs may not be written to disk, except here is where I'm writing my logs`, which feels contradictory.

Reply:
You have shamed me in my laziness. As penance I further engorged this PR with new functionality, leading to this behavior:

```console
$ spin up --from examples/http-rust
Logging component stdio to "examples/http-rust/.spin/logs/"

Serving [...]
```
```console
$ spin up --from examples/rust-key-value
Logging component stdio to "examples/rust-key-value/.spin/logs/"
Storing key-value data in "examples/rust-key-value/.spin/sqlite_key_value.db"

Serving [...]
```
```console
$ spin up --from-registry ttl.sh/lann-rust-key-value:1m
Using in-memory key-value store; data will not be saved!

Serving [...]
```

Reply:
Is that... is that only printing the K/V message if the app actually uses the default key-value store?  BEAUTIFUL.  Love this.

Reply:
Indeed! Well it just looks for the `key_value_stores` key in the manifest, but that should be approximately the same.